plugins/inputs/modbus/modbus.go

```diff
@@ -349,10 +349,10 @@ func validateFieldContainers(t []fieldContainer, n string) error {
 		}
 
 		// check address
-		if len(item.Address) == 0 || len(item.Address) > 2 {
+		if len(item.Address) != 1 && len(item.Address) != 2 && len(item.Address) != 4 {
 			return fmt.Errorf("invalid address '%v' length '%v' in '%s' - '%s'", item.Address, len(item.Address), n, item.Name)
 		} else if n == cInputRegisters || n == cHoldingRegisters {
-			if (len(item.Address) == 1 && len(item.ByteOrder) != 2) || (len(item.Address) == 2 && len(item.ByteOrder) != 4) {
+			if (len(item.Address) == 1 && len(item.ByteOrder) != 2) || (len(item.Address) == 2 && len(item.ByteOrder) != 4) || (len(item.Address) == 4 && len(item.ByteOrder) != 8) {
```

Comment:
I'd like to see some simplification of the logic here. Eg: `if len(item.Address) * 2 != len(item.ByteOrder) .... `

Reply:
Makes a lot of sense! Hopfully solved with my commit below... :-)